smart-contract-feature-development.md

```diff
@@ -0,0 +1,94 @@
+# Purpose
+
+The purpose of this document is to provide guidance on how develop smart contracts without the need for feature branches.
+
+# Summary
+
+Feature branches result in a lot of effort spent keeping the feature branch up to date. We want to have modular code
+that enables us to easily compose features together such that we can get code mainlined early and toggle it off when
+running in production.
+
+# Problem Statement + Context
+
+We want to move away from a world with long lived feature branches.
+To do so, we need to have some flexibility when it comes to composition of features.
+This is easy in off-chain code as the cost is minimal for toggling off certain features but still having them in the source code.
+This is not the case for Solidity. This document exists to outline how to think about composition of features for Solidity code.
+When the feature is ready to be shipped, the feature specific contract should be merged into the base contract.
+
+# Alternatives Considered
```

Comment:
We probably should have a section for feature branches as an alternative also, given the "Feature branches result in a lot of effort spent keeping the feature branch up to date" aspect of the summary. I would be curious to hear if there are learnings you can share from past attempts with feature branches for solidity development